### PR TITLE
EW-13175: EdgeKV CLI should output a proper error message for bad retention input

### PR DESCRIPTION
**Pull request by slukic0**
- Validate retention period duration before calling API
- Fix a few broken tests in `ekv-services.test.ts`

### CODE REVIEW

File: src/edgekv/ekv-helper.ts [hunk 1]
```diff
@@ -356,6 +356,10 @@ export function getDateDifference(date) {
 
 export function convertDaysToSeconds(days: number) {
     if (!isNaN(days) && days >= 0) {
+        if (days != 0 && (days < 1 || days > 3650)){
```
Comment (swathimr) [inline src/edgekv/ekv-helper.ts:359]: I think both these conditions (days != 0 && days < 1) are already covered in "days >= 0". Would recommend moving the max check to the outer loop.
if (!isNaN(days) && (days >= 0  || days > 3650))

Comment (slukic0) [reply src/edgekv/ekv-helper.ts:359]: @swathimr Any value between 0 and 1 is invalid (eg 0.5), but the value 0 is still a valid retention period (unlimited). Not sure if your suggestion would still handle that case correctly?

I was trying to make the if statement clear by saying that the range is only required if `days` is not zero. Perhaps I could add a code comment to clarify this?

Comment (ahsankamal) [reply src/edgekv/ekv-helper.ts:359]: @swathimr 
We get the same error message from the create namespace edgeKV API for very large value of retention period. Thanks @slukic0 for finding it. 
At the CLI level, we are just outputting the same error message in console that we get from the create namespace edgeKV API. So, if we add this limit condition at the CLI level and later limit for retention period gets changed at the API level, we would need to change it at the CLI level again. Does it make sense to just fix it at the API level or is it fine to add limit conditions in CLI codebase also?

**Review by ahsankamal — APPROVED** [review]

File: src/edgekv/ekv-helper.ts (hunk 1 above)
Comment (swathimr) [reply src/edgekv/ekv-helper.ts:359]: talked offline and sorted it out.

**Review by swathimr — APPROVED** [review]

**Review by bmatthew — APPROVED** [review]

**Pull request merged**